File: app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Product extends Model {
    use HasFactory;

    protected $fillable = [
        "user_id",
        "slug",
        "name",
        "category",
        "price",
        "quantity",
        "image",
        "description"
    ];

    public function response(){
        return [
            "user" => $this->user->response(),
            "slug" => $this->slug,  
            "name" => $this->name,
            "category" => $this->category,
            "price" => $this->price,
            "quantity" => $this->quantity,
            "image" => $this->image,
            "description" => $this->description,
        ];
    }

    public function user(): BelongsTo {
        return $this->belongsTo(User::class);
    }
}

File: model/entity/product.go
package entity

import "time"

type Product struct {
	Id          string    `gorm:"column:id;primaryKey" json:"id"`
	Slug        string    `gorm:"column:slug;unique" json:"slug"`
	Name        string    `gorm:"column:name" json:"name"`
	Price       float32   `gorm:"column:price" json:"price"`
	Quantity    int       `gorm:"column:quantity" json:"quantity"`
	Image       *string   `gorm:"column:image" json:"image"`
	Description string    `gorm:"column:description" json:"description"`
	UserId      string    `gorm:"column:user_id" json:"user_id"`
	CreatedAt   time.Time `gorm:"column:created_at" json:"created_at"`
	UpdatedAt   time.Time `gorm:"column:updated_at" json:"updated_at"`

	User *User `gorm:"foreignKey:UserId;references:Id" json:"user"`
}
